solar_system_wave3.rb

```diff
@@ -0,0 +1,261 @@
+# assignment to learn about classes and methods
+# gives info about planets in a solar system, allows user to create new planet class
+# ada week 2, ruby basics
+
+class SolarSystem
+  attr_accessor :planets, :name, :age
+  #don't know if accessor is right here, or in Planet, but I didn't have the bandwidth to play with public/private methods...
+
+  def initialize(name, age, planets)
+    @name = name
+    @age = age
+    @planets =  planets
+
+
+  end
+
+# method to add a new planet
+  def add_planet(input)
+    @planets << input
+  end
+
+# method to return one piece of data from a planet
+  def planet_method(planet_name, method)
+    case method
+    when :name
+      info = planets_by_name(planet_name).name
+    when :mass
+      info = planets_by_name(planet_name).mass
+    when :index_from_sun
+      info = planets_by_name(planet_name).index_from_sun
+    when :circumference
+      info = planets_by_name(planet_name).circumference
+    when :color
+      info = planets_by_name(planet_name).color
+    when :year_length
+      info = planets_by_name(planet_name).year_length
+    when :distance_from_sun
+      info = planets_by_name(planet_name).distance_from_sun
+    when :all_facts
+      info = planets_by_name(planet_name).all_facts
+    else
+      puts "error"
+      return
+    end
+
+    return info
+  end
+
+  # doesn't work because it calls methods that call it?
+  # def planets_by_name(name)
+  #   index = all_planets(:name).index(name.capitalize.to_s)
+  #   return @planets[index]
+  # end
+
+  # method to access the planet class by name so as to use planet methods on it
+  def planets_by_name(name)
+    planet_names = []
+    @planets.each do |planet_class|
```

Comment:
I really like the idea of making this its own method. Very concise bit of functionality.

You're implementation is a little odd to me though - do you really need to pull out the planet names? Why not:

```ruby
@planets.each do |planet|
  if planet.name == name
    return planet
  end
  return nil
end
```